modules/nats/options.go

```diff
@@ -16,12 +17,21 @@ func defaultOptions() options {
 	}
 }
 
-// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
-var _ testcontainers.ContainerCustomizer = (*CmdOption)(nil)
+// Compiler check to ensure that CmdOption and ConfigFile implements the
+// testcontainers.ContainerCustomizer interface.
+var (
+	_ testcontainers.ContainerCustomizer = (*CmdOption)(nil)
+	_ testcontainers.ContainerCustomizer = (*ConfigFile)(nil)
+)
 
 // CmdOption is an option for the NATS container.
 type CmdOption func(opts *options)
 
+// ConfigFile optionally pass a configuration file into NATS container.
+type ConfigFile struct {
```

Comment:
I'm not sure if we need an extra type but instead leverage the `testcontainers.CustomiseRequestOption`. As an example, please take the postgres module: https://github.com/testcontainers/testcontainers-go/blob/main/modules/postgres/postgres.go#L66-L79, it will make the code simpler

Also, could you add the new option to the docs, at `docs/modules/nats.md`, right after the cmd-arguments entry 🙏 ? It will land into the [docs site for NATS](https://golang.testcontainers.org/modules/nats/#cmd-arguments)

Reply:
Rewritten code according postgres module: https://github.com/testcontainers/testcontainers-go/pull/2905/commits/a3636bbb3b0946ea20f2270d704a419faf20b7cb

Updated docs: https://github.com/testcontainers/testcontainers-go/pull/2905/commits/9a9225f150a516d4e49da0703bd7831ce19ae66a

Not sure about a proper since annotation, so I've omitted it from PR.

Reply:
Oh you're right about the "since". Let me add a suggestion now